sdks/uniswapx-sdk/src/utils/dutchBlockDecay.ts

```diff
@@ -0,0 +1,124 @@
+import { BigNumber } from "ethers";
+
+import { NonlinearDutchDecay } from "../order";
+/*
+These functions mimic the smart contract functions as closely as possible to ensure that the same results are produced.
+Essentially Solidity translated to TypeScript.
+*/
+function locateArrayPosition(
+	curve: NonlinearDutchDecay,
+	targetValue: number
+): [number, number] {
+	const relativeBlocks = curve.relativeBlocks;
+	// eslint-disable-next-line prefer-const
+	let prev = 0;
+	// eslint-disable-next-line prefer-const
+	let next = 0;
+
+	while (next < curve.relativeAmounts.length) {
```

Comment:
Just a heads up, per Eric's rec I've updated the contract to use a for loop instead.

Reply:
updated to align!